### PR TITLE
Rename module highlight token

### PR DESCRIPTION
**Pull request by jonatanklosko**
See #15.

Updating the highlight token from `@type` to `@module`, so that module names get the expected color on GitHub.

### CODE REVIEW

File: src/scanner.cc [hunk 1]
```diff
@@ -56,7 +56,10 @@ bool is_inline_whitespace(int32_t c) {
 }
 
 bool is_newline(int32_t c) {
-  return c == '\n';
+  // Note: this implies \r\n is treated as two line breaks,
+  // but in our case it's fine, since multiple line breaks
+  // make no difference
+  return c == '\n' || c == '\r';
```
Comment (jonatanklosko) [inline src/scanner.cc:62]: The integration tests found a new edge case when parsing [axon example](https://github.com/elixir-nx/axon/blob/main/examples/vision/horses_or_humans.exs) :)
